Makefile

```diff
@@ -1,31 +1,67 @@
+.PHONY: all clean test check install uninstall release
+
 all:
 	@dune build
 
+clean:
+	@git clean -fX
+
 test:
 	@dune runtest
 
+check: test
+
 install:
 	@dune install
 
 uninstall:
 	@dune uninstall
 
-check: test
-
-.PHONY: clean all check test install uninstall
-
-clean:
-	dune clean
-
 # To make a release:
-# + check that [make] and [make test] succeed
-# + check that everything has been committed
+# + check that [make test] succeeds
+# + check that everything has been committed and pushed
 # + check that the CI has succeeded
-# + check that [opam lint] is happy
 # + make sure that the package is not pinned: [opam pin remove cppo]
-# + create a tag, e.g. [git tag vX.Y.Z]
-#   (the tag should not be annotated, as it shows up in [cppo -version])
-# + run [make publishX.Y.Z]
+# + run [make release VERSION=X.Y.Z]
 
-publish%:
-	opam publish --tag=v$* -v $* ocaml-community/cppo
+release:
+# Make sure the current version can be compiled.
+	@ make clean
+	@ make test
+# Check if this is the master branch.
+	@ if [ "$$(git symbolic-ref --short HEAD)" != "master" ] ; then \
```

Comment:
Should the checks to see if we're on the master branch, everything has been committed, etc. be before the make test so you fail early?

Reply:
Fine with me! I have exchanged the checks.